app/src/main/java/org/kiwix/kiwixmobile/zim_manager/local_file_transfer/LocalFileTransferActivity.java

```diff
@@ -0,0 +1,437 @@
+ * 2) After handshake, starting the files transfer using {@link SenderDeviceAsyncTask} on the sender
+ * device and {@link ReceiverDeviceAsyncTask} files receiving device
+ */
+@SuppressLint("GoogleAppIndexingApiWarning")
+public class LocalFileTransferActivity extends AppCompatActivity implements
+    WifiDirectManager.Callbacks {
+
+  // Not a typo, 'Log' tags have a length upper limit of 25 characters
+  public static final String TAG = "LocalFileTransferActvty";
+  public static final int REQUEST_ENABLE_LOCATION_SERVICES = 1;
+  private static final int PERMISSION_REQUEST_CODE_COARSE_LOCATION = 1;
+  private static final int PERMISSION_REQUEST_CODE_STORAGE_WRITE_ACCESS = 2;
+
+  @Inject SharedPreferenceUtil sharedPreferenceUtil;
+  @Inject AlertDialogShower alertDialogShower;
+
+  @BindView(R.id.toolbar_local_file_transfer) Toolbar actionBar;
+  @BindView(R.id.text_view_device_name) TextView deviceName;
+  @BindView(R.id.progress_bar_searching_peers) ProgressBar searchingPeersProgressBar;
+  @BindView(R.id.list_peer_devices) ListView listViewPeerDevices;
+  @BindView(R.id.text_view_empty_peer_list) TextView textViewPeerDevices;
+  @BindView(R.id.recycler_view_transfer_files) RecyclerView filesRecyclerView;
+
+  private boolean isFileSender = false;    // Whether the device is the file sender or not
+
+  private @NonNull WifiDirectManager wifiDirectManager = new WifiDirectManager(this);
+
+  private ArrayList<FileItem> filesForTransfer = new ArrayList<>();
+  private FileListAdapter fileListAdapter;
+
+  private List<WifiP2pDevice> availablePeerDevices = new ArrayList<WifiP2pDevice>();
+  private boolean hasSenderStartedConnection = false;
+
+  @Override
+  protected void onCreate(@Nullable Bundle savedInstanceState) {
+    KiwixApplication.getApplicationComponent().activityComponent()
+        .activity(this)
+        .build()
+        .inject(this);
+    setTheme(sharedPreferenceUtil.nightMode() ? R.style.Theme_AppCompat_DayNight_NoActionBar : R.style.AppTheme);
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.activity_local_file_transfer);
+    ButterKnife.bind(this);
+
+    /*
+     * Presence of file Uris decides whether the device with the activity open is a sender or receiver:
+     * - On the sender device, this activity is started from the app chooser post selection
+     * of files to share in the Library
+     * - On the receiver device, the activity is started directly from within the 'Get Content'
+     * activity, without any file Uris
+     * */
+    Intent filesIntent = getIntent();
+    ArrayList<Uri> fileUriArrayList;
+    fileUriArrayList = filesIntent.getParcelableArrayListExtra(Intent.EXTRA_STREAM);
+    isFileSender = (fileUriArrayList != null && fileUriArrayList.size() > 0);
+
+    setSupportActionBar(actionBar);
+    actionBar.setNavigationIcon(R.drawable.ic_close_white_24dp);
+    actionBar.setNavigationOnClickListener(new View.OnClickListener() {
```

Comment:
can be replaced with lambda, there should be an autocomplete action

Reply:
[Done](https://github.com/kiwix/kiwix-android/pull/1276/commits/4e7a2e03c2957184e4ac59c2d901d958fd752092#diff-e1d59ae6e15dcb502e125fb6e243d15dR120)